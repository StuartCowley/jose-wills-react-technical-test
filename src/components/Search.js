import React, { useState } from "react";
import "../styles/search.css";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState("");

  const handleInputChange = (event) => setValue(event.target.value);

  const handleSubmit = (event) => {
    event.preventDefault()
    setSearchResults(getImages(value));
  };

  return (
    <form onSubmit={handleSubmit}>
      <input className="search__input" type="search" placeholder="Enter image keywords here" onChange={handleInputChange} />
      <input className="search__button" type="submit" value="Search" />
    </form>
  );
};

export default Search;
