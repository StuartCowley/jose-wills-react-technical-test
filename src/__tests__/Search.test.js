import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<Search />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("has a text input", () => {
        render(<Search />);

        const searchInput = screen.getByPlaceholderText("Enter image keywords here");

        expect(searchInput).toBeInTheDocument();
    });
});

