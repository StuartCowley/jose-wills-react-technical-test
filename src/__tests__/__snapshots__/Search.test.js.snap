// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form>
    <input
      class="search__input"
      placeholder="Enter image keywords here"
      type="search"
    />
    <input
      class="search__button"
      type="submit"
      value="Search"
    />
  </form>
</DocumentFragment>
`;
